src/middlewared/middlewared/plugins/disk.py

```diff
@@ -275,6 +277,52 @@ def geli_testkey(self, pool, passphrase):
                         return False
         return True
 
+    @private
+    def geli_setkey(self, dev, key, slot=GELI_KEY_SLOT, passphrase=None, oldkey=None):
+        cp = subprocess.run([
+            'geli', 'setkey', '-n', str(slot),
+        ] + (
+            ['-J', passphrase] if passphrase else ['-P']
+        ) + ['-K', key] + (
+            ['-k', oldkey] if oldkey else []
+        ) + [dev], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        if cp.stderr:
+            raise CallError(f'Unable to set passphrase on {dev}: {cp.stderr.decode()}')
+
+    @private
+    def geli_delkey(self, dev, slot=GELI_KEY_SLOT, force=False):
+        cp = subprocess.run([
+            'geli', 'delkey', '-n', str(slot),
+        ] + (
+            ['-f'] if force else []
+        ) + [dev], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        if cp.stderr:
+            raise CallError(f'Unable to delete key {slot} on {dev}: {cp.stderr.decode()}')
+
+    @private
+    def geli_passphrase(self, pool, passphrase, rmrecovery=False):
+        """
+        Set a passphrase in a geli
+        If passphrase is None then remove the passphrase
+        """
+        if passphrase:
```

Comment:
If we care about `None`, perhaps this check should be a bit more specific with `if passphrase is None`

Reply:
We dont, empty string is to be treated as none.